registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/dao/PacketInfoDao.java

```diff
@@ -1,695 +1,672 @@
+				.getAbisRequestIDsbasedOnIdentity(latestTransactionId, requestType);
+		for (AbisRequestEntity abisRequestEntity : abisRequestEntities) {
+			abisResponseEntities.addAll(abisResponseRepository.getAbisResponseIDs(abisRequestEntity.getId().getId()));
+		}
+		abisResponseDto.addAll(PacketInfoMapper.convertAbisResponseEntityListToDto(abisResponseEntities));
+		return abisResponseDto;
+	}
+
+	/**
+	 * Gets the abis response records.
+	 *
+	 * @param abisRefId
+	 *            the abis ref id
+	 * @param latestTransactionId
+	 *            the latest transaction id
+	 * @param requestType
+	 *            the request type
+	 * @return the abis response records
+	 */
+	public List<AbisResponseDto> getAbisResponseRecords(String abisRefId, String latestTransactionId,
+			String requestType) {
+		List<AbisResponseEntity> abisResponseEntities = new ArrayList<>();
+		List<AbisResponseDto> abisResponseDto = new ArrayList<>();
+		List<AbisRequestEntity> abisRequestEntities = abisRequestRepository.getInsertOrIdentifyRequest(abisRefId,
+				latestTransactionId, requestType);
+		for (AbisRequestEntity abisRequestEntity : abisRequestEntities) {
+			abisResponseEntities.addAll(abisResponseRepository.getAbisResponseIDs(abisRequestEntity.getId().getId()));
+		}
+		abisResponseDto.addAll(PacketInfoMapper.convertAbisResponseEntityListToDto(abisResponseEntities));
+		return abisResponseDto;
+	}
+
+	/**
+	 * Gets the abis response detailed records.
+	 *
+	 * @param abisResponseDto
+	 *            the abis response dto
+	 * @return the abis response detailed records
+	 */
+	public List<AbisResponseDetDto> getAbisResponseDetailedRecords(AbisResponseDto abisResponseDto) {
+		List<AbisResponseDetDto> abisResponseDetDtoList = new ArrayList<>();
+		List<AbisResponseDetEntity> abisResEntity = abisRequestRepository
+				.getAbisResponseDetails(abisResponseDto.getId());
+		abisResponseDetDtoList.addAll(PacketInfoMapper.convertAbisResponseDetEntityListToDto(abisResEntity));
+		return abisResponseDetDtoList;
+	}
+
+	/**
+	 * Gets the abis requests by bio ref id.
+	 *
+	 * @param bioRefId
+	 *            the bio ref id
+	 * @return the abis requests by bio ref id
+	 */
+	public List<AbisRequestEntity> getAbisRequestsByBioRefId(String bioRefId) {
+		return abisRequestRepository.getAbisRequestsByBioRefId(bioRefId, "INSERT");
+	}
+
+	/**
+	 * Gets the abis processed requests app code by bio ref id.
+	 *
+	 * @param bioRefId
+	 *            the bio ref id
+	 * @param requestType
+	 *            the request type
+	 * @param processed
+	 *            the processed
+	 * @return the abis processed requests app code by bio ref id
+	 */
+	public List<String> getAbisProcessedRequestsAppCodeByBioRefId(String bioRefId, String requestType,
+			String processed) {
+		return abisRequestRepository.getAbisProcessedRequestsAppCodeByBioRefId(bioRefId, requestType, processed);
+
+	}
+
+	public List<RegistrationStatusEntity> getWithoutStatusCode(List<String> matchedRegIds, String statusCode) {

```

Comment:
if its getWithoutStatusCode then why we have statucode as a parameter?

Reply:
We are calling getWithoutStatusCode with REJECTED statusCode ,It will pick up records without REJECTED status code